scripts/tank_cmd.py

```diff
@@ -1230,7 +1254,148 @@ def run_engine_cmd(log, pipeline_config_root, context_items, command, using_cwd,
         return tank_command.run_action(log, tk, ctx, command, args)
 
 
+def _extract_args(cmd_line, *args):
+    """
+    Extract specified arguments and sorts them. Also removes them from the
+    command line.
+
+    :param cmd_line: Array of command line arguments.
+    :param args: Argument list to extract
+
+    :returns: A list of key, value tuples representing each arguments.
+    """
+    # Find all instances of each argument in the command line
+    found_args = []
+    for cmd_line_token in cmd_line:
+        for arg in args:
+            if cmd_line_token.startswith("--%s=" % arg):
```

Comment:
I don't think this would work with the following command lines:

```
tank --login=foo --password=bar unregister_folders --all
tank --login=foo --password=bar setup_project --force
```

The way the tank command works, is that it has the following syntax:

```
tank context_options action action_params
```

For example 

```
tank shot 123 my_command param1 param2 param3
```

We want to pass everything in the command line after my_command as an arg string to my_command so that the command can do the processing it needs to. 

The general approach I would take would be to look for specific parameters in the arg string, if found then consume them, remove them from the command line. You can see how it is done (very simplistically) for `--debug`. As part of this code review, i looked at how feasible it would be to move to optionparser but it seems it doesn't work well for this two-pass approach we have.

How about something along the lines of 

```python
params = {}
new_args = []
keywords = ["login", "password", "script_key", "script_name"]
for arg in args:
   # look for each of our keywords in the command line
   found = False
   for keyword in keywords:
       k = "--%s=" % keyword
       if arg.startswith(k):
           keyword_val = arg[len(k):]
           params[keyword] = keyword_val
           found = True
           break
    if not found:
        # keep for later
        new_args.append(arg)
return (params, new_args)
```

(The validation of the param values would then happen outside of the method)